jwst/ramp_fitting/ramp_fit.py

```diff
@@ -422,175 +459,164 @@ def create_output_models(input_model, number_of_integrations, save_opt, total_co
         The number of frames dropped at the beginning of every integration
+
     int_times : None
         Not used
 
     Returns
     -------
     new_model.data : 2-D float32
         The output final rate of each pixel
+
     new_model.dq : 2-D DQflag
         The output pixel dq for each pixel
+
     new_model.var_poisson : 2-D float32
         The variance in each pixel due to Poisson noise
+
     new_model.var_rnoise : 2-D float32
-        The variance in each piel due to read noise
+        The variance in each pixel due to read noise
+
     new_model.err : 2-D float32
         The output total variance for each pixel
+
     int_data : 3-D float32
         The rate for each pixel in each integration
+
     int_dq : 3-D float32
         The pixel dq flag for each integration
+
     int_var_poisson : 3-D float32
         The variance of the rate for each integration due to Poisson noise
+
     int_var_rnoise : 3-D float32
         The variance of the rate for each integration due to read noise
+
     int_err : 3-D float32
         The total variance of the rate for each integration
+
     int_int_times : 3-D
         The total time for each integration
+
     opt_slope : 4-D float32
         The rate of each segment in each integration
+
     opt_sigslope : 4-D float32
         The total variance of the rate for each pixel in each segment of each integration
+
     opt_var_poisson : 4-D float32
         The Poisson variance of the rate for each pixel in each segment of each integration
+
     opt_var_rnoise : 4-D float32
         The read noise variance of the rate for each pixel in each segment of each integration
+
     opt_yint : 4-D float32
         The y-intercept for each pixel in each segment of each integration
+
     opt_sigyint : 4-D float32
         The variance for each pixel in each segment of each integration
+
     opt_pedestal : 4-D float32
         The zero point for each pixel in each segment of each integration
+
     opt_weights : 4-D float32
         The weight of each pixel to use in combining the segments
+
     opt_crmag : 4-D float32
         The magnitude of each CR in each integration
+
     actual_segments : int
         The actual maximum number of segments in any integration
+
     actual_CRs : int
         The actual maximum number of CRs in any integration
     """
-    tstart = time.time()
+    # Package image data into a RampModel
+    input_model = RampModel()
 
-    # Get needed sizes and shapes
-    n_int = data.shape[0]
-    nreads = data.shape[1]
-    nrows = data.shape[2]
-    ncols = data.shape[3]
-    imshape = (nrows, ncols)
-    cubeshape = (nreads,) + imshape
-    # Save original shapes for writing to log file, as these may change for MIRI
-    orig_nreads = nreads
-    orig_cubeshape = cubeshape
-    if dropframes1 is None:    # set to default if missing
-        dropframes1 = 0
-        log.debug('Missing keyword DRPFRMS1, so setting to default value of 0')
+    input_model.data = data
```

Comment:
This is a wrapper I used because I changed the function prototype for `ols_ramp_fit` to `ols_ramp_fit_single`.  This is an intermediate fix to continue refactoring the OLS code without having to refactor the multi-processing code right now, also.  Ultimately, this unpacking and repacking will be refactored out.